### PR TITLE
Add curseforge service and sample bundle

### PR DESCRIPTION
**Pull request by MelanX**

### CODE REVIEW

File: nodecg-io-curseforge/extension/curseforgeClient.ts [hunk 1]
```diff
@@ -0,0 +1,1034 @@
...
+export class CurseAddon {
+    private readonly curse: CurseForge;
+    readonly addonId: number;
+    readonly info: CurseAddonInfo;
+
+    constructor(curse: CurseForge, addonId: number, info: CurseAddonInfo) {
+        this.curse = curse;
+        this.addonId = addonId;
+        this.info = info;
+    }
```
Comment (daniel0611) [inline nodecg-io-curseforge/extension/curseforgeClient.ts:198]: nit: You can declare class members right in the constructor parameter list to make this simpler:
```suggestion
    constructor(private readonly curse: CurseForge, public readonly addonId: number, public readonly info: CurseAddonInfo) {}
```

File: nodecg-io-curseforge/extension/curseforgeClient.ts [hunk 2]
```diff
@@ -0,0 +1,1034 @@
...
+    index?: number;
+    pageSize?: number;
+    searchFilter?: string;
+    sectionId?: number;
+    sort?: number;
+};
+
+export type CurseFeaturedAddonsQuery = {
+    GameId: number;
+    addonsIds?: number[];
```
Comment (daniel0611) [inline nodecg-io-curseforge/extension/curseforgeClient.ts:929]: According to the [docs](https://twitchappapi.docs.apiary.io/#/reference/0/get-featured-addons/get-featured-addons/200?mc=reference%2F0%2Fget-featured-addons%2Fget-featured-addons%2F200) this should be `addonIds` (without the s after addons)
```suggestion
    addonIds?: number[];
```

**Review by daniel0611 — CHANGES_REQUESTED** [review]
Thanks for this in advance! However there are some open points:

**Comment by MelanX** [comment]
I hope everything is fine now

**Review by daniel0611 — APPROVED** [review]
Thanks!

**Pull request merged**